### PR TITLE
Add Torrent Link modal

### LINKED ISSUE #78: + button should show prompt
Make the `+` button on the header bar show a prompt for a torrent id string (magnet link, http/fs path to .torrent, etc.).

### PR DESCRIPTION
**Pull request by dcposch**
image

Fixes #78

### CODE REVIEW

File: renderer/index.js [hunk 1]
```diff
@@ -157,6 +163,9 @@ function dispatch (action, ...args) {
   if (action === 'addTorrent') {
     addTorrent(args[0] /* torrent */)
   }
+  if (action === 'addFilePrompt') {
+    ipcRenderer.send('showOpenTorrentFile')
```
Comment (feross) [inline renderer/index.js]: Can we make these names consistent?

File: renderer/views/app.js [hunk 2]
```diff
@@ -33,16 +34,33 @@ function App (state, dispatch) {
...
+    var isOSX = process.platform === 'darwin'
     // Hide the header on Windows/Linux when in the player
     if (isOSX || state.url !== 'player') {
       return Header(state, dispatch)
     }
   }
 
+  function getModal () {
+    if (state.modal) {
+      /* dynamically load modals, don't require() them upfront */
```
Comment (feross) [inline renderer/views/app.js]: This comment is out-of-date

File: renderer/index.js [hunk 3]
```diff
@@ -489,7 +508,7 @@ function toggleTorrent (torrentSummary) {
 function deleteTorrent (torrentSummary) {
   var infoHash = torrentSummary.infoHash
   var torrent = getTorrent(infoHash)
-  torrent.destroy()
+  if (torrent) torrent.destroy()
```
Comment (feross) [inline renderer/index.js]: good call. was seeing errors about this

File: main/menu.js [hunk 4]
```diff
@@ -79,44 +108,25 @@ function getMenuTemplate () {
...
           type: 'separator'
         },
         {
-          label: (function () {
-            if (process.platform === 'darwin') return 'Close Window'
-            else return 'Close'
-          })(),
+          label: process.platform === 'darwin'
+            ? 'Close Window'
+            : 'Close',
```
Comment (feross) [inline main/menu.js]: Good cleanup

**Comment by feross** [comment]
LGTM

File: renderer/index.js (hunk 1 above)
Comment (dcposch) [inline renderer/index.js]: done

**Comment by feross** [comment]
:+1: :100:

**Pull request merged**